src/main/java/org/broadinstitute/hellbender/utils/recalibration/BaseRecalibrationEngine.java

```diff
@@ -219,23 +219,26 @@ private void updateRecalTablesForRead( final ReadRecalibrationInfo recalInfo ) {
         final ReadCovariates readCovariates = recalInfo.getCovariatesValues();
         final NestedIntegerArray<RecalDatum> qualityScoreTable = recalTables.getQualityScoreTable();
 
-        for( int offset = 0; offset < read.getLength(); offset++ ) {
+        final int nCovariates = covariates.size();
+        final int nSpecialCovariates = covariates.numberOfSpecialCovariates();
+        for( int offset = 0, n= read.getLength(); offset < n; offset++ ) {
```

Comment:
spacing fixed.
technically, that call is outside the loop - it's in the initializer and thus called only once. I moved it out to clarify